src/lib/http-agent.js

```diff
@@ -28,7 +28,8 @@ const DEFAULT_HTTPS_PORT = 443
 // 50 seconds
 const AGENT_PORT_TIMEOUT = 50
 
-const getAgent = async ({ httpProxy, certificateFile, exit }) => {
+// a stub utility is used in tests for exit
+const getAgent = async ({ httpProxy, certificateFile, exit = exit_ }) => {
```

Comment:
There might be some opportunity to improve how this function is being tested.

At the moment, we mock `exit`, although we don't really need to. Instead, we could split `getAgent()` into two functions:
  - `tryGetAgent({ httpProxy, certificateFile })`, returning an object either `{ agent }` (success) or `{ agent, warning: 'warning message' }` or `{ error: 'error message' }`
  - `getAgent({ httpProxy, certificateFile })`, the top-level function, would call `tryGetAgent()`. If it returns: 
     - an `error`, it would `log()` it with `NETLIFYDEVERR` and call `exit(1)`
     - a `warning`, it would `log()` it with `NETLIFYDEVWARN` but not `exit()`
     - an `agent`, it would return it

This approach would have the benefits of:
  - Decoupling logging/exiting from the main actual logic
  - Removing the need to mock `exit()` in the unit tests, since those could test `tryGetAgent()` instead

What do you think?

Reply:
I think it's better to check function responses rather than the behaviour of process control and logging utilities.

One thing is that it'd create an export which won't be used anywhere except for the tests. Is that okay?

Reply:
> One thing is that it'd create an export which won't be used anywhere except for the tests. Is that okay?

I think this is a side effect of unit testing (as opposed to integration testing). Since we are testing at a small scale, we sometimes need to hook into specific functions, i.e. create exports for them just for testing purpose.

Reply:
Gotcha!